core-bundle/src/Resources/contao/pages/PageRegular.php

```diff
@@ -45,7 +47,15 @@ public function getResponse($objPage, $blnCheckRequest=false)
 	{
 		$this->prepare($objPage);
 
-		return $this->Template->getResponse($blnCheckRequest);
+		$response = $this->Template->getResponse($blnCheckRequest);
+		$responseContext = System::getContainer()->get(ResponseContextAccessor::class)->getResponseContext();
+
+		if ($responseContext instanceof WebpageResponseContext)
```

Comment:
why are we checking the instanceof here?

Reply:
Because we are accessing methods of it afterwards. Any service that gets the current response context must test if it's the correct one and either do nothing (in case of our future JSON-LD implementation) or throw an exception if it was unexpected. Nobody should be doing `new PageRegular()` without setting the context previously. If I did not check here, they would get a PHP error saying that you cannot call `getMetaDescription()` on `null` ;)

Reply:
ah so this is not to check the interface, but to check against NULL values (not having a request context)?

Reply:
It makes sure that we have a `WebpageResponseContext`, yes. In a new page controller, creating the context would be something of the first things you would do in your invoication method and then you know it must be there. Here, however, we have a "legacy" page type. Its controller resides in `FrontendIndex::renderPage()` which means that in theory someone could've called `new PageRegular()` instead of properly using the Symfony controller. I want to make sure that people notice this error. Calling `new PageRegular()` manually has always been wrong, imho. You should've called the controller for page `xy` :)